### PR TITLE
Likes and matches logic

### PR DESCRIPTION
**Pull request by RuslanKamenev**

### CODE REVIEW

File: backend/database/migrations/2022_03_28_145042_create_user_location_table.php [hunk 1]
```diff
@@ -0,0 +1,28 @@
+<?php
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class CreateUserLocationTable extends Migration
```
Comment (OsipchukArtem) [inline backend/database/migrations/2022_03_28_145042_create_user_location_table.php:7]: class CreateUserLocationsTable extends Migration

File: backend/database/migrations/2022_03_28_145042_create_user_location_table.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+{
+    public function up()
+    {
+        Schema::create('user_location', function (Blueprint $table) {
```
Comment (OsipchukArtem) [inline backend/database/migrations/2022_03_28_145042_create_user_location_table.php:11]: Schema::create('user_locations', function (Blueprint $table) {

File: backend/database/migrations/2022_03_28_145042_create_user_location_table.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+                ->references('id')
+                ->on('users');
+        });
+    }
+
+    public function down()
+    {
+        Schema::dropIfExists('user_location');
+    }
+}
```
Comment (OsipchukArtem) [inline backend/database/migrations/2022_03_28_145042_create_user_location_table.php:28]: rename 
backend/database/migrations/2022_03_28_145042_create_user_location_table.php 
into
backend/database/migrations/2022_03_28_145042_create_user_locations_table.php

File: backend/app/Http/Middleware/UserActivity.php [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+use App\Models\User;
+use Closure;
+use Illuminate\Http\Request;
+use Illuminate\Support\Carbon;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Cache;
+
+class UserActivity
+{
+    /**
```
Comment (OsipchukArtem) [inline backend/app/Http/Middleware/UserActivity.php:14]: delete comments

**Review by OsipchukArtem — APPROVED** [review]

File: backend/app/Actions/Matches/GetAllMatchesResponse.php [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    }
+
+    public function getUsers(): array
+    {
+        return $this->usersList;
+    }
+
+    public function distanceToUser($userId): int
+    {
+        return intval($this->distanceToUsers[$userId]);
```
Comment (RomanMahola) [inline backend/app/Actions/Matches/GetAllMatchesResponse.php:24]: (int)

File: backend/app/Http/Controllers/Api/Matches/MatchesController.php [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+            ->execute(
+                new GetAllUsersListRequest(
+                    $request->all(),
+                    $request->ip()
+                )
+            );
+
+        $usersData = [];
+        foreach ($result->getUsers() as $key => $user) {
+            if ($result->statusRequest() && ($userRepository->isUserOnline($user) === false)) {
```
Comment (RomanMahola) [inline backend/app/Http/Controllers/Api/Matches/MatchesController.php:78]: I DON`T LIKE IT

File: backend/app/Repositories/UserParameterNew/UserParameterNewRepository.php [hunk 7]
```diff
@@ -17,4 +29,38 @@ public function save(UserParameterNew $userParameter): bool
...
+            ->limit(100)
+            ->get()
+            ->all();
+    }
+
+    public function getUsersById(array $usersId): array
+    {
+        return UserParameterNew::select('user_parameters_new.*', 'users.name as name')
+            ->join('users', 'users.id', '=', 'user_parameters_new.user_id')
+            ->whereIn('user_id', $usersId)->get()->all();
```
Comment (RomanMahola) [inline backend/app/Repositories/UserParameterNew/UserParameterNewRepository.php:59]: user_parameters_new.user_id

File: frontend/src/bundles/main/components/UserDetailsForm.vue [hunk 8]
```diff
@@ -28,8 +28,7 @@
             rules="required|min:3"
           >
             <v-text-field
-              v-if="userInfo"
-              v-model="userInfo.name"
+              v-model="localUserAdditionalInfo.name"
```
Comment (RomanMahola) [inline frontend/src/bundles/main/components/UserDetailsForm.vue:31]: localUserAdditionalInfo = { ...localUserAdditionalInfo, name: value }

**Review by RomanMahola — APPROVED** [review]

**Pull request merged**